jenkins/upgrade_python_requirements.py

```diff
@@ -10,8 +10,7 @@
 @click.command()
 @click.option(
     '--sha',
-    help="Sha of the merge commit to base the new PR off of",
-    required=True,
+    help="DEPRECATED: Ignored, current SHA is read from repo now",
```

Comment:
As noted, I just don't want us to leave this very long, because it seems risky.

Reply:
Could you help me understand what is risky?  Or is it worth having a ticket to fully remove it later on?

Reply:
It looks like the commit ID that is passed in is currently always the HEAD commit. This change is to ignore that input and always use HEAD.  I believe Robert's concern is that if a script passes in a value that wasn't HEAD, the PR creator would make the wrong PR.

The cleanup work is recorded as ARCHBOM-1632.